js_modules/dagit/packages/core/src/app/AppTopNav.tsx

```diff
@@ -14,16 +14,18 @@ import {WebSocketStatus} from './WebSocketProvider';
 
 interface Props {
   searchPlaceholder: string;
+  rightOfSearchBar?: React.ReactNode;
 }
 
-export const AppTopNav: React.FC<Props> = ({children, searchPlaceholder}) => {
+export const AppTopNav: React.FC<Props> = ({children, rightOfSearchBar, searchPlaceholder}) => {
   const history = useHistory();
 
   return (
     <AppTopNavContainer>
       <Box flex={{direction: 'row', alignItems: 'center', gap: 16}}>
         <AppTopNavLogo />
```

Comment:
Something more flexible but probably overkill for now would be if we had a prop

`getLeftItems` thats a function like so:
```
(logo: React.ReactNode, searchBar: React.ReactNode): React.ReactNode => {
  return (
    <>
       {logo}
       {searcBar}
       <Tag>Free Trial ends May 31</Tag>
    </>
  );
}
```

Reply:
Yeah, I'm intrigued by this possibility.

Especially since the only non-children prop right now is `searchPlaceholder`, and it looks like we're just using the same value in both places anyway. And then `children` is basically just "stuff on the right".

```
<AppTopNav
  left={({logo, search}) => (
    <>
      {logo}
      {search}
      <Tag>...</Tag>
    </>
  )}
  right={({tabs}) => (
    <>
      {tabs}
      <Deployment ... />
      <Identity ... />
      /* or in OS, SettingsLink instead */
    </
  )}
/>
```